p2p/src/net/mock/transport/impls/stream_adapter/wrapped_transport/wrapped_listener.rs

```diff
@@ -43,11 +45,12 @@ impl<S: StreamAdapter<T::Stream>, T: TransportSocket> TransportListener<S::Strea
 {
     async fn accept(&mut self) -> Result<(S::Stream, T::Address)> {
         loop {
+            let accept_new = self.handshakes.len() < MAX_CONCURRENT_HANDSHAKES;
             tokio::select! {
                 handshake = HandshakeFut::<S, T>(&mut self.handshakes) => {
                     return Ok(handshake);
                 }
-                accept_res = self.listener.accept() => {
+                accept_res = self.listener.accept(), if accept_new => {
```

Comment:
What happens now if a client connects and there are already 100 handshakes in progress?

Reply:
I think the TCP connection will be in pending state and once some handshake is complete we will process it normally (I have not tested this).

Reply:
Please add a test for this, and also add a test to check what happens locally when the other peer times out.